### PR TITLE
NodeStateTest and systemd targets

### PR DESCRIPTION
**Pull request by DunderRoffe**
[GDP-3](https://at.projects.genivi.org/jira/browse/GDP-3)
Add the systemd targets: focused, unfocused and lazy. These targets are
required by the specifiction of the GENIVI lifecycle to start the HMI and
provide the last user context.

[GDP-554](https://at.projects.genivi.org/jira/browse/GDP-554)
Install NodeStateTest on the GDP platfrom. It is used to demonstrate that
the NodeSateManager setup works on the platform.

### CODE REVIEW

File: meta-genivi-dev/recipes-core/systemd/systemd_%.bbappend [hunk 1]
```diff
@@ -2,9 +2,17 @@ FILESEXTRAPATHS_append := ":${THISDIR}/${PN}"
...
+    file://lazy.target \
+    "
 
 do_install_append() {
     install -d ${D}${sysconfdir}
     install -m 0644 ${WORKDIR}/system.conf ${D}${sysconfdir}/systemd/system.conf
+
+    install -m 0644 ${WORKDIR}/focused.target   ${D}/lib/systemd/system/focused.target
+    install -m 0644 ${WORKDIR}/unfocused.target ${D}/lib/systemd/system/unfocused.target
+    install -m 0644 ${WORKDIR}/lazy.target      ${D}/lib/systemd/system/lazy.target
```
Comment (chbae) [inline meta-genivi-dev/recipes-core/systemd/systemd_%.bbappend]: ${D}/lib/systemd/system -> ${D}${systemd_system_unitdir}

File: meta-genivi-dev/recipes-core/systemd/systemd_%.bbappend [hunk 2]
```diff
@@ -2,9 +2,17 @@ FILESEXTRAPATHS_append := ":${THISDIR}/${PN}"
...
+    file://unfocused.target \
+    file://lazy.target \
+    "
 
 do_install_append() {
     install -d ${D}${sysconfdir}
     install -m 0644 ${WORKDIR}/system.conf ${D}${sysconfdir}/systemd/system.conf
+
+    install -m 0644 ${WORKDIR}/focused.target   ${D}/lib/systemd/system/focused.target
+    install -m 0644 ${WORKDIR}/unfocused.target ${D}/lib/systemd/system/unfocused.target
```
Comment (chbae) [inline meta-genivi-dev/recipes-core/systemd/systemd_%.bbappend]: ${D}/lib/systemd/system -> ${D}${systemd_system_unitdir}

File: meta-genivi-dev/recipes-core/systemd/systemd_%.bbappend [hunk 3]
```diff
@@ -2,9 +2,17 @@ FILESEXTRAPATHS_append := ":${THISDIR}/${PN}"
...
+    file://focused.target \
+    file://unfocused.target \
+    file://lazy.target \
+    "
 
 do_install_append() {
     install -d ${D}${sysconfdir}
     install -m 0644 ${WORKDIR}/system.conf ${D}${sysconfdir}/systemd/system.conf
+
+    install -m 0644 ${WORKDIR}/focused.target   ${D}/lib/systemd/system/focused.target
```
Comment (chbae) [inline meta-genivi-dev/recipes-core/systemd/systemd_%.bbappend]: ${D}/lib/systemd/system -> ${D}${systemd_system_unitdir}

**Review by chbae — CHANGES_REQUESTED** [review]

**Comment by chbae** [comment]
NodeStateManager: install NodeStateTest -> Why do you install test binary into target?

**Review by zeenix — CHANGES_REQUESTED** [review]
As usual, @chbae quickly takes care of review of patch itself so I can focus on commit log. :) Actually this is a pretty good one and I don't need to complain much about syntax:

* There is no such thing as "lifecycle" technically speaking so we really shouldn't refer to it as such. Especially the last sentence makes it sound like "GDP-lifecycle" is the name of a software that we'll be integrating later, which is not at all the case here.

* You're missing jira ticket reference.

* Please don't break lines for new sentences. This is supposed to be normal English text so only good reason to break to new line are
  * starting a new paragraph (in which case you need an empty line in between the paragraphs)
  * You hit 74 chars limit.

**Review by zeenix — CHANGES_REQUESTED** [review]
Some rationale in details of why this change is needed, would be nice. Also missing the jira reference.

**Comment by DunderRoffe** [comment]
@chbae The test binary is included as a tool to demonstrate that NSM works on the target.

**Comment by DunderRoffe** [comment]
Sure @zeenix this pull-request was a bit stressed as I wanted to get it up before the team meeting

**Review by zeenix — APPROVED** [review]

**Comment by zeenix** [comment]
@chbae looks good to me now, once you're satisfied please approach the changes and merge.

**Review by zeenix — CHANGES_REQUESTED** [review]
On second thought, why bbappend here and not do this in the recipe (in meta-ivi)? We already have a lot of bbappend and we want to reduce them.

**Comment by DunderRoffe** [comment]
Because that would affect any other project using meta-ivi. I am not sure that every other project wants to build the test binary

**Comment by zeenix** [comment]
@DunderRoffe good point. :)

**Review by zeenix — APPROVED** [review]

**Comment by gunnarx** [comment]
Sorry to come in late with this review but I have preferences and pointers to make:

Licenses:

- Generally apply MPLv2 (GENIVI default license) unless you have a good reason. Even when there is a good reason to apply (L)GPL, the GENIVI license policy only allows LGPL v2.1 for straight-up libraries.

- Since you didn't avoid LGPL naturally here, I assume an explanation might help:  In my personal view LGPLv2.1 is confusing and possibly even legally ambiguous when applied to non-library code (despite that it mentions "other program" in the introduction, it is very clear the rest of it has been written with libraries in mind).   The GENIVI license review team used similar reasoning when deciding on the policy.

- Here, I'd personally prefer a permissive license or CC0 for the really uninteresting metadata (which I would consider systemd unitfiles to be).  I'd recommend that we decide and document a guideline for all "system configuration" type data in GDP and make sure we are at least consistent.

**Comment by gunnarx** [comment]
I also react to the bbappend, but the one for systemd.  The files are part of defining the GDP system, not really part of building systemd itself.  I wonder if there is a more natural place to put them.

**Comment by zeenix** [comment]
@chbae are you concerns satisfied now? If so, please do approve the changes.

@gunnarx how about yours? Please set "request changes" in your reviews if that's the case so it's easy to track the status of PR. Thanks.

**Review by chbae — APPROVED** [review]

**Comment by chbae** [comment]
Sorry for late review.

**Pull request merged**